ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -575,7 +579,107 @@ private JoinNode buildJoin(final Join root, final String prefix, final boolean i
     );
   }
 
+  private void verifyJoin(final Join root, final PlanNode left, final PlanNode right) {
+    if (left.getNodeOutputType() == DataSourceType.KSTREAM) {
+      if (right.getNodeOutputType() == DataSourceType.KTABLE) {
+
+        // stream-table join detected
+
+        if (root.getInfo().getType().equals(JoinType.OUTER)) {
+          throw new KsqlException("Invalid join type:"
+              + " full-outer join not supported for stream-table joins.");
+        }
+
+        // added outer check to make `multi-joins.json` pass
+        // the test uses expressions -> need to follow up on this issue
+        if (root.getInfo().getRightJoinExpression() instanceof ColumnReferenceExp) {
```

Comment:
Need to fix this, to be able to process `BinaryArithmeticExpressions` and maybe others?

Reply:
`joinOnNonKeyAttribute()` should always return true if the expression is not a `ColumnReferenceExp`, right? Why don't we just pass the entire expression in and let `joinOnNonKeyAttribute()` perform the check?

Reply:
My understanding was different. If we do `t1 JOIN t2 ON t1.key = t2.key + 1` it's still a PK-join not a FK-join right?However, ``1 JOIN t2 ON t1.a = t2.key + 1` would be a FK-joins. For both cases, it seems we would repartition `t2` onto new key `key + 1`?

I am not sure what expression we support atm and what parsing on the expression we would need to do to cover all cases? I _would_ assume that the expression can only contain a single "field/columnName"? So maybe it's sufficient to parse the full expression-tree and extract the single field?

Reply:
If it is a `ColumnReferenceExp` then I think it would not be arithmetic, like `t2.key + 1` atm (just reading through the sqlBase here).

Reply:
Yes. If it's a `ColumnReferenceExp` we do the check, but if not we skip it, but IMHO we need to do the check even if it's not a `ColumnReferenceExp`. (For now, I just put this guard in place to avoid that `multi-joins.json` pass, but it's just a workaround to make it pass -- we need to properly address those cases.)